osvimdriver/service/resourcedriver.py

```diff
@@ -315,6 +369,22 @@ def __determine_create_status(self, request_id, stack_id, stack_status):
         logger.debug('Stack %s has stack_status %s, setting status in response to %s', stack_id, stack_status, create_status)
         return create_status
 
+    def __determine_adopt_status(self, request_id, stack_id, stack_status):                
+        if self.adopt_config.skip_status_check:
+            logger.debug('Status check for adopt of stack id: %s will be skipped', stack_id)            
+            return STATUS_COMPLETE
+
+        if stack_status in [OS_STACK_STATUS_CREATE_IN_PROGRESS, OS_STACK_STATUS_ADOPT_IN_PROGRESS, OS_STACK_STATUS_RESUME_IN_PROGRESS, OS_STACK_STATUS_CHECK_IN_PROGRESS]:
+            adopt_status = STATUS_IN_PROGRESS
+        elif stack_status in self.adopt_config.adoptable_status_values:  
+            adopt_status = STATUS_COMPLETE
+        elif stack_status in [OS_STACK_STATUS_CREATE_FAILED, OS_STACK_STATUS_ADOPT_FAILED, OS_STACK_STATUS_CHECK_FAILED, OS_STACK_STATUS_SUSPEND_IN_PROGRESS, OS_STACK_STATUS_SUSPEND_COMPLETE]:
+            adopt_status = STATUS_FAILED
+        else:
```

Comment:
Is there an opportunity for confusion here, in the sense that the user would assume that the `adoptable_status_values` constitute a _complete_ set of statuses that indicate `STATUS_COMPLETE`? What happens if `adoptable_status_values` includes e.g. OS_STACK_STATUS_CREATE_IN_PROGRESS - should values in `adoptable_status_values`  override what the driver is explicitly checking here?

Reply:
True, I have rearranged the order of the checks so that the `adoptable_status_values` are checked first